pkg/services/sqlstore/migrations/ualert/ualert.go

```diff
@@ -738,3 +741,42 @@ func (u *upgradeNgAlerting) updateAlertmanagerFiles(orgId int64, migrator *migra
 func (u *upgradeNgAlerting) SQL(migrator.Dialect) string {
 	return "code migration"
 }
+
+// CheckUnifiedAlertingEnabledByDefault determines the final status of unified alerting, if it is not enabled explicitly.
+// Checks table `alert` and if it is empty, then it changes UnifiedAlerting.Enabled to true. Oterwise, it sets the flag to false.
+// After this method the status of alerting should be determined.
+// Note: this is not a real migration but a step that other migrations depend on.
+// TODO Delete when unified alerting is enabled by default unconditionally (Grafana v9)
+func CheckUnifiedAlertingEnabledByDefault(migrator *migrator.Migrator) error {
+	if migrator.Cfg.UnifiedAlerting.Enabled != nil {
+		return nil
+	}
+	resp := &struct {
+		Count int64
+	}{}
+	exist, err := migrator.DBEngine.IsTableExist("alert")
+	if err != nil {
+		return fmt.Errorf("failed to access the database to determine alerting status: %w", err)
+	}
```

Comment:
I do not think that it makes sense to fail here If the table does not exist. It will fail somewhere else or not at all. I think we can safely assume that the absence of the table mean that UA is supposed to be enabled

Reply:
I don't think that's possible - even with unified alerting enabled on a fresh install, the `alert` table should exist as it otherwise would mean that the migration never ran or there is an inconsistency between the migration log and what's in the database. 

I do think this is an unexpected behaviour but I'm fine with not returning an error and changing it for a logline + return if you feel strongly about it.

From reading the code (and because there was no comment attached to it) it seemed as case we missed.

Reply:
I left a comment that we tolerate the absence of the table.